ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ClusterDetailsPage.tsx

```diff
@@ -1,26 +1,113 @@
-import React from 'react';
-import { useHistory, useParams } from 'react-router-dom';
-import { Button, PageSection } from '@patternfly/react-core';
+import React, { useCallback } from 'react';
+import { generatePath, useParams } from 'react-router-dom';
+import {
+    Alert,
+    Breadcrumb,
+    BreadcrumbItem,
+    Divider,
+    Flex,
+    Label,
+    LabelGroup,
+    PageSection,
+    Skeleton,
+    Title,
+} from '@patternfly/react-core';
 
-import { complianceEnhancedCoveragePath } from 'routePaths';
+import BreadcrumbItemLink from 'Components/BreadcrumbItemLink';
+import PageTitle from 'Components/PageTitle';
+import useRestQuery from 'hooks/useRestQuery';
+import { getComplianceProfilesStats } from 'services/ComplianceResultsStatsService';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+
+import {
+    coverageProfileClustersPath,
+    coverageClusterDetailsPath,
+} from './compliance.coverage.routes';
+import ProfilesToggleGroup from './ProfilesToggleGroup';
 
 function ClusterDetailsPage() {
-    const history = useHistory();
-    const { profileName } = useParams();
+    const { clusterId, profileName } = useParams();
+
+    const fetchProfilesStats = useCallback(
+        () => getComplianceProfilesStats(clusterId),
+        [clusterId]
+    );
+    const {
+        data: clusterProfileData,
+        loading: isLoadingClusterProfileData,
+        error: clusterProfileDataError,
+    } = useRestQuery(fetchProfilesStats);
```

Comment:
Was playing around with the idea of adding an optional default value to the `useRestQuery` hook. Doing that would help with the issue of data potentially being `undefined`. 

```javascript
const {
    data: clusterProfileData,
    loading: isLoadingClusterProfileData,
    error: clusterProfileDataError,
} = useRestQuery(fetchProfilesStats, {scanStats: [], totalCount: 0});
```

You could type the `defaultValue` to also be the `ReturnType` that way typescript will complain if the object doesn't match the shape of the requests return type.

This would prevent us from having to do this:
* `Scanned by: ${clusterProfileData?.totalCount} profiles`
* `profiles={clusterProfileData?.scanStats || []}`